lib/orbit/ocs/repo/entities.ex

```diff
@@ -0,0 +1,267 @@
+defmodule Orbit.Ocs.Entities do
+  @moduledoc """
+  Responsible for processing parsed OCS messages, and writing appropriate changes
+  to the Trains and Trips in the database.
+  Also provides querying of relevant entities for the current service date.
+  """
+  require Logger
+
+  alias Orbit.Ocs.Train
+  alias Orbit.Ocs.Trip
+  alias Orbit.RailLine
+  alias Orbit.Repo
+
+  alias Orbit.Ocs.Message
+
+  alias Orbit.Ocs.Message.{
+    TschAsnMessage,
+    TschConMessage,
+    TschDelMessage,
+    TschDstMessage,
+    TschLnkMessage,
+    TschNewMessage,
+    TschOffMessage,
+    TschRldMessage,
+    TschTagMessage
+  }
+
+  import Ecto.Query
+
+  @type entities() :: [Trip.t()]
+
+  @spec query_latest(DateTime.t()) :: entities()
+  def query_latest(current_datetime \\ DateTime.utc_now()) do
+    # TODO: What filtering/sorting makes sense for this query?
+    service_date = Util.Time.service_date_for_utc_datetime(current_datetime)
+    Repo.all(from(trip in Trip, where: trip.service_date == ^service_date))
+  end
+
+  @spec to_utc(DateTime.t() | nil) :: DateTime.t() | nil
+  defp to_utc(datetime) do
+    if datetime == nil do
+      nil
+    else
+      DateTime.shift_zone!(datetime, "Etc/UTC")
+    end
+  end
+
+  @spec apply_changes(Message.t()) :: [{:ok, any} | {:error, any()}]
+  def apply_changes(%TschNewMessage{} = message) do
+    %Trip{
+      service_date: service_date(message.timestamp),
+      uid: message.trip_uid,
+      prev_uid: message.prev_trip_uid,
+      next_uid: message.next_trip_uid,
+      route: message.ocs_route_id,
+      rail_line: RailLine.from_ocs_transitline(message.transitline),
+      trip_type: message.trip_type,
+      scheduled_departure: to_utc(message.sched_dep),
+      scheduled_arrival: to_utc(message.sched_arr),
```

Comment:
question: I'm perplexed why these `to_utc`s are necessary. either:
1. It's zoned and it's ET, in which case the database will handle it
2. It's zoned and it's UTC, in which case the database handle it
3. It's naive, in which case `shift_zone!` is not enough

?